src/sentry/static/sentry/app/views/stream.jsx

```diff
@@ -69,8 +77,24 @@ var Stream = React.createClass({
   },
 
   routeDidChange() {
-    this._poller.disable();
-    this.fetchData();
+    var currentQuery = this.context.router.getCurrentQuery();
+
+    var filter = {};
+    if (currentQuery.bookmarks) {
+      filter = { bookmarks: "1" };
+    } else if (currentQuery.assigned) {
+      filter = { assigned: "1" };
+    }
+
+    var query = (currentQuery.hasOwnProperty("query")) ? currentQuery.query : this.props.defaultQuery;
+
+    this.setState({
+      filter: filter,
+      query: query
+    }, function() {
+      this._poller.disable();
```

Comment:
I mean do we need to do function() {}.bind(this) (or simply () => {}) to ensure this is reflecting the correct scope?